### PR TITLE
6436 modernise third party fat bucket

### PR DESCRIPTION
**Pull request by benjamin-confino**
Modernise fat tests by merging both tests that use logs, and moving the test that uses HTTP to fat servlet client

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi12/fat/tests/HibernateSearchTest.java [hunk 1]
```diff
@@ -10,65 +10,70 @@
...
+    @ClassRule
+    public static RepeatTests r = EERepeatTests.with(SERVER_NAME, EE8);
 
     @BeforeClass
     public static void setUp() throws Exception {
 
-        WebArchive hibernateSearchTest = ShrinkWrap.create(WebArchive.class, "hibernateSearchTest.war")
+        //Hibernate Search Test
+        WebArchive hibernateSearchTest = ShrinkWrap.create(WebArchive.class, HIBERNATE_SEARCH_APP_NAME+".war")
                         .addPackages(true, "cdi.hibernate.test")
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.third.party_fat/fat/src/com/ibm/ws/cdi12/fat/tests/HibernateSearchTest.java:69]: If possible please could you switch from a hard coded package name to a real package from one of the classes? That way it still works if things get moved.

File: dev/com.ibm.ws.cdi.third.party_fat/test-applications/hibernateSearchTest.war/src/cdi/hibernate/test/web/SimpleTestServlet.java [hunk 2]
```diff
@@ -49,23 +61,19 @@
...
             throws ServletException, IOException {
-        final PrintWriter pw = response.getWriter();
-        bean.getString();
-        pw.println("<h1>JPA Simple Test Case</h1>");
-        pw.println("<hr>");
         try {
-            // We trigger this method to indirectly call the field bridge.
-            CommonTestCode.populate(request, response, em, tx);
+            PrintWriter pw = new PrintWriter(new ByteArrayOutputStream());
+            CommonTestCode.populate(pw, em, tx);
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.third.party_fat/test-applications/hibernateSearchTest.war/src/cdi/hibernate/test/web/SimpleTestServlet.java:71]: Do you want to check the contents of the PrintWriter after the populate call? A PrintWriter probably isn't the best way to do it any more anyway?

File: dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ShrinkHelper.java [hunk 3]
```diff
@@ -128,6 +137,13 @@ private static boolean serverOnly(DeployOptions[] options) {
      */
     public static void exportAppToServer(LibertyServer server, Archive<?> a, DeployOptions... options) throws Exception {
         exportToServer(server, "apps", a, options);
+
+        String appName = a.getName();
+        if (shouldValidate(options)) {
+            String installedAppName = (appName.endsWith(".war") || appName.endsWith(".ear"))
+                        ? appName.substring(0, appName.length() - 4) : appName;
+            server.addInstalledAppForValidation(installedAppName);
+        }
```
[the remark this answers is not in the document]
Comment (tevans78) [reply dev/fattest.simplicity/src/com/ibm/websphere/simplicity/ShrinkHelper.java:146]: Ben, did you check this?

**Review by tevans78 — APPROVED** [review]

**Comment by benjamin-confino** [comment]
#run-libby-bot

**Comment by wraschke** [comment]
This PR causes a hard build break in the LAOS continuous builds. See https://ibm-cloud.slack.com/archives/C30NGUQ9E/p1599673924125800 for details.

**Pull request merged**